models/tiegcm/model_mod.f90

```diff
@@ -160,82 +161,52 @@ module model_mod
 integer, parameter :: VT_ORIGININDX   = 5 ! ... file of origin
 integer, parameter :: VT_STATEINDX    = 6 ! ... update (state) or not
 
-character(len=obstypelength) :: variable_table(max_num_variables, max_num_columns)
-
-! include_vTEC = .true.  vTEC must be calculated from other vars
-! include_vTEC = .false. just ignore vTEC altogether
-
-logical  :: include_vTEC = .true.
-logical  :: include_vTEC_in_state = .false.
-
-! IMPORTANT: 1 D model parameters (e.g., F107) are read in from "tiegcm.nml"
-! (note "estimate_f10_7" option is still under
-! development by Tomoko Matsuo as of June 24, 2011)
+character(len=obstypelength) :: variable_table(MAX_NUM_VARIABLES, MAX_NUM_COLUMNS)
 
-real(r8)        :: f10_7
 type(time_type) :: state_time ! module-storage declaration of current model time
 
-integer                :: model_size
-real(r8), allocatable  :: ens_mean(:)
-
-! FOR NOW OBS LOCATIONS ARE EXPECTED GIVEN IN HEIGHT [m],
-! AND SO VERTICAL LOCALIZATION COORDINATE IS *always* HEIGHT
-! (note that gravity adjusted geopotential height (ZG)
-!  read in from "tiegcm_s.nc" *WARNING* ZG is 'cm', DART is mks)
-
-real(r8), allocatable, dimension(:,:,:) :: ZG
+integer(i8)           :: model_size ! the state vector length
+integer :: nfields  ! number of tiegcm variables in DART state
+! global domain id to be used by routines in state_structure_mod
+integer :: domain_id(3) ! restart, secondary, calculate
+integer, parameter :: RESTART_DOM = 1
+integer, parameter :: SECONDARY_DOM = 2
+integer, parameter :: CONSTRUCT_DOM = 3
+
+! lon and lat grid specs. 2.5 degree or 5 degree grid
+real(r8)  :: bot_lon        = MISSING_R8
+real(r8)  :: top_lon        = MISSING_R8
+real(r8)  :: delta_lon      = MISSING_R8
+real(r8)  :: bot_lat        = MISSING_R8
+real(r8)  :: top_lat        = MISSING_R8
+real(r8)  :: delta_lat      = MISSING_R8
+integer   :: zero_lon_index = MISSING_I
+
+
+! Obs locations are expected to be given in height [m],
```

Comment:
maybe adjust this comment?  it looks like model_interpolate also supports obs on model levels?  it's probably true that only height localization is supported, but asking for obs values on levels seems like it works?

Reply:
done.